packages/next/next-server/server/render.tsx

```diff
@@ -651,7 +651,9 @@ export async function renderToHTML(
     // use replace to allow rendering directly to body in AMP mode
     html = html.replace(
```

Comment:
Would it be better to just not use replace? I think the dollar sign inception is confusing.

Something like this, but with a hoisted constant and maybe re-used index:
```js
html.substring(0, html.indexOf("__NEXT_AMP_RENDER_TARGET__")) +
  `<!-- __NEXT_DATA__ -->${docProps.html}` +
  html.substring(
    html.indexOf("__NEXT_AMP_RENDER_TARGET__") +
      "__NEXT_AMP_RENDER_TARGET__".length
  );
```

Reply:
Updated